openhands/server/listen.py

```diff
@@ -790,20 +790,18 @@ async def security_api(request: Request):
 
 
 @app.get('/api/zip-directory')
```

Comment:
I'm curious what happens if we make a new endpoint with the same logic, but give it a `.zip` file extension. Does the browser treat it as a normal download, with a status bar?

Reply:
I tested this - changing to ".zip" does not seem to affect the download - at least in chrome. The only way I can see of getting a progress bar right now is to create a background task, because we are actually monitoring 3 things:
* Creating the zip file
* Transferring the zip file out of the runtime
* Transferring the zip file to the client